### PR TITLE
Fixed empty assignment field in IPAddress detail view.

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #4819
# What's Changed

### No assignments

image

### 1 VM Interface assignment - 0 Interface assignments

image

### 1 Interface assignment - 0 VM Interface assignments

image

### 1 Interface assignment - 1 VM Interface assignment

image

### 2 Interface assignments - 1 VM Interface assignment

image

### 2 Interface assignments - 0 VM interface assignments

image

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/ipam/templates/ipam/ipaddress.html [hunk 1]
```diff
@@ -95,14 +95,17 @@ <h1>
...
-                        {% if object.assigned_object %}
-                            {{ object.assigned_object.parent|hyperlinked_object }} ({{ object.assigned_object }})
-                        {% else %}
-                            <span class="text-muted">&mdash;</span>
-                        {% endif %}
+                        {% with interface_count=object.interfaces.count %}
+                            {% if interface_count == 1 %}
+                                {{ object.interfaces.first.parent|hyperlinked_object }} ({{ object.interfaces.first }})
+                            {% elif interface_count > 1 %}
+                                <a href="{% url 'ipam:ipaddress_interfaces' pk=object.pk %}">{{ interface_count }}</a>
```
Comment (glennmatthews) [inline nautobot/ipam/templates/ipam/ipaddress.html:104]: Perhaps:

```suggestion
                                <a href="{% url 'ipam:ipaddress_interfaces' pk=object.pk %}">{{ interface_count }} interfaces</a>
```

**Review by glennmatthews — COMMENTED** [review]
Don't we need to account for `object.vm_interfaces` as well?

**Comment by HanlinMiao** [comment]
> Don't we need to account for `object.vm_interfaces` as well?

Yeahhh, maybe an additional table row for `vm_interfaces`?

**Comment by gsnider2195** [comment]
> Don't we need to account for `object.vm_interfaces` as well?

There's no tab for vm interfaces but we can add that as unscoped work.

**Comment by glennmatthews** [comment]
In 1.x, `object.assigned_object` covered both Interface and VMInterface assignments. IMO fixing #4819 requires addressing both cases as well.

File: nautobot/ipam/templates/ipam/ipaddress_vm_interfaces.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        </div>
+    </form>
+    {% table_config_form vm_interface_table %}
+{% endblock content %}
+
+{% block javascript %}
+    {{ block.super }}
+    <script src="{% static 'js/connection_toggles.js' %}?v{{ settings.VERSION }}"></script>
+    <script src="{% static 'js/interface_filtering.js' %}?v{{ settings.VERSION }}"></script>
+    <script src="{% static 'js/tableconfig.js' %}?v{{ settings.VERSION }}"></script>
```
Comment (glennmatthews) [inline nautobot/ipam/templates/ipam/ipaddress_vm_interfaces.html:31]: More stuff that'll need to be touched in #4892, just FYI.

**Review by glennmatthews — APPROVED** [review]
👏 🚀

**Review by HanlinMiao — APPROVED** [review]

**Pull request merged**